features/0434-outofband/README.md

```diff
@@ -0,0 +1,417 @@
+# Aries RFC 0434: Out of Band Protocol 1.0
+
+- Authors: [Ryan West](ryan.west@sovrin.org), [Daniel Bluhm](daniel.bluhm@sovrin.org), Matthew Hailstone, [Stephen Curran](swcurran@cloudcompass.ca), [Sam Curren](sam@sovrin.org), [George Aristy](george.aristy@securekey.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2020-03-01
+- Status Note: This RFC extracts the `invitation` messages from the [DID Exchange](../../features/0023-did-exchange/README.md) protocol (and perhaps [Connection](../../features/0160-connection-protocol/README.md)), and replaces the combined `present_proof/1.0/request` combined with the `~service` decorator to define an ephemeral (connection-less) challenge.
+- Supersedes: Invitation Message in [0160-Connections](https://github.com/hyperledger/aries-rfcs/blob/9b0aaa39df7e8bd434126c4b33c097aae78d65bf/features/0160-connection-protocol/README.md#0-invitation-to-connect) and Invitation Message in [0023-DID-Exchange](https://github.com/hyperledger/aries-rfcs/blob/9b0aaa39df7e8bd434126c4b33c097aae78d65bf/features/0023-did-exchange/README.md#0-invitation-to-exchange).
+- Start Date: 2020-03-01
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol)
+- Protocol Name: Out of Band
+- Version: 1.0
+- URI: `http://didcomm.org/out-of-band/VER/MsgType`
+
+## Summary
+
+The Out Of Band protocol is used when you wish to engage with another agent and you don't have a DIDComm connection to use for the interaction.
+
+## Motivation
+
+The use of the `invitation` in the [Connection](../../features/0160-connection-protocol/README.md) and [DID Exchange](../../features/0023-did-exchange/README.md) protocols has been relatively successful, but has some shortcomings, as follows.
+
+### Connection Reuse
+
+A common pattern we have seen in the early days of Aries agents is a user with a browser getting to a point where a connection is needed between the website's (enterprise) agent and the user's mobile agent. A QR invitation is displayed, scanned and a protocol is executed to establish a connection. Life is good!
+
+However, with the current invitation processes, when the same user returns to the same page, the same process is executed (QR code, scan, etc.) and a new connection is created between the two agents. There is no way for the user's agent to say "Hey, I've already got a connection with you. Let's use that one!"
+
+We need the ability to reuse a connection.
+
+### Connection Establishment Versioning
+
+In the existing Connections and DID Exchange `invitation` handling, the _inviter_ dictates what connection establishment protocol all _invitee_'s will use. A more sustainable approach is for the _inviter_ to offer the _invitee_ a list of support protocols and allow the _invitee_ to use one that it supports. This pattern enables the community deprecation of protocol versions.
```

Comment:
I think this paragraph is out of alignment with the description of protocol version negotiation in https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0003-protocols/semver.md#recipient-rules. It has never been the case that inviters dictate versions as far as the specs are concerned; it's just the case that nobody in the community has implemented the version negotiation logic we already described.

If you believe what I am asserting, then the value of this as a rationale is lower, and the advisability of sending a list of supported protocols is questionable.

Reply:
I think we would all like to reduce the instances of negotiation in the connection establishment process. We could support it, but avoiding it would be MUCH better. Negotiation of protocols once a connection is established is perfectly reasonable, but I think it's a bad idea on creating the connection.

Reply:
Please go read what was written. It's not a back-and-forth negotiation. It's a "here's the version of the protocol I want to use, as embodied by the first message I send" followed by "here's the second message in the protocol, but notice that i've degraded the version, so you're stuck with what I picked" sort of negotiation. It's not chatty at all. It introduces zero negotiation messages.

Reply:
In the existing RFCs, the inviter picks the invitation to send - either Connection or DID Exchange, since they can only send one. By taking "invitation" out of those, the receiver can respond to either during the next transition. After that, the value goes away, particularly if there is only version changes going forward.

I think you are proposing that in the Out of Band, invitation we should not specify any protocol and the receiver can respond with any protocol--presumably limited to those meant to establish a connection?  I think if we drop it, we have to be very clear in the RFC on the intention.

It does not feel right to me, but what do others think?

While as you say, the "value is reduced", I think it is still worth doing, but it does come at the expense of the length of the message (and hence, more likely URL shortening).

Reply:
I don't see an issue with listing the supported *protocols* in the out-of-band invitation. I don't see an issue with including the supported *versions* as well, but I don't see that as a critical point.

I think what @dhh1128 is doing is attacking the assertion in the highlighted paragraph that including the *version* "enables the community to deprecate protocol versions" when in fact such a mechanism is described in RFC0003. There's a point there, but this out-of-band invitation would be just another way to achieve the same goal (deprecation of protocol versions), although solely contained to the bootstrapping phase.

Reply:
Thanks, George.  I've removed that line.

Reply:
Still reading the conversation, but I'm a fan of listing the protocols. I think it's unlikely that we won't have the need for supporting different connection protocols.